### DISCUSSION TITLE
IDLE3 on MacOS Big Sur

### DISCUSSION THREAD

**Discussion by cdr-chakotay** [Homebrew/discussions#435] [Everyday usage] [answered]
Recently installed Python 3.9.1 via brew.
Whilst python is working fine, IDLE is not.
Launching ILDE with:
`$ idle3`
results in:        
`macOS 11 or later required !
zsh: abort      idle3`
It also throws an Mac OS Exception, indicating TCL-Panic

Funfact, i am on MacOS 11.1 and a non homebrew version i tested afterwards has not shown this behaviour.

Can anybody reproduce or solve my issue?

**Comment by TXBDan** [comment]
I also have this problem.

**Comment by carlocab** [comment, marked as answer]
This should be fixed by https://github.com/Homebrew/homebrew-core/pull/67378.

**Reply by TXBDan** [reply]
I just did a fresh homebrew install on my M1 MacBook Air. I did a native ARM install and it worked properly, no errors.I then simply did "brew install python" which installed 3.9.1. Again, no errors. I'm running idle3 from "/opt/homebrew/opt/python3/bin". 

I am a noob so plenty of ways I could have screwed something up, but no dice so far. Same error as the OP.

**Reply by carlocab** [reply]
Yes, but that's because the fix hasn't been merged yet. What would be surprising is if, either:
1. the fix hasn't been merged, but you can no longer reproduce the error; or,
2. the fix has been merged but you still encounter the error.

**Reply by TXBDan** [reply]
Ok, I'll delete my crash report post since the fix seems to be under control and a fix is on the way.  In the mean time I'll find a different IDE I guess.

**Reply by cdr-chakotay** [reply]
> Ok, I'll delete my crash report post since the fix seems to be under control and a fix is on the way. In the mean time I'll find a different IDE I guess.

Usually i'm using PyCharm in the free (community) edition but for fast script editing or to try something out without overhead i just love using IDLE. 
Maybe PyCharm is an option for you, too.

**Reply by carlocab** [reply]
The fix should be ready in the next day or so. Thanks for your patience!

**Comment by carlocab** [comment]
Fix has been merged, please do `brew update && brew upgrade python@3.9`. @TXBDan
